src/Foundatio/Utility/ScheduledTimer.cs

```diff
@@ -61,7 +63,8 @@ public class ScheduledTimer : IDisposable {
 
                 _logger.Trace(() => $"Scheduling next: delay={delay}");
 
-                _timer.Change(delay, Timeout.Infinite);
+                DisposeTimer();
+                _timer = SystemClock.Instance.Schedule(TimeSpan.FromMilliseconds(delay), async () => await RunCallbackAsync());
```

Comment:
Also thinking more on this, could be a ton of allocations / work creating a new timer all the time.